### PR TITLE
[Fusion] Improve Inlining Rewriter

### PR DESCRIPTION
**Pull request by michaelstaib**

### CODE REVIEW

File: src/HotChocolate/Fusion-vnext/test/Fusion.Execution.Tests/Planning/InlineFragmentOperationRewriterTests.cs [hunk 1]
```diff
@@ -237,4 +236,45 @@ fragment Product on Product {
...
+            """
+            query($skip: Boolean!) {
+                productById(id: 1) {
+                    ... Product
+                }
+            }
+
+            fragment Product on Product {
+                id
+                name @include(if: $skip)
```
Comment (michaelstaib) [inline src/HotChocolate/Fusion-vnext/test/Fusion.Execution.Tests/Planning/InlineFragmentOperationRewriterTests.cs:257]: @tobias-tengler I have reworked the field merging with what we discussed. It needs more tests and especially for edge cases.

Comment (michaelstaib) [reply src/HotChocolate/Fusion-vnext/test/Fusion.Execution.Tests/Planning/InlineFragmentOperationRewriterTests.cs:257]: We actually could handle static includes and skips at this level already. This way we would not need to deal with them in the planning process.

File: src/HotChocolate/Fusion-vnext/src/Fusion.Execution/Planning/InlineFragmentOperationRewriter.cs [hunk 2]
```diff
@@ -224,10 +243,239 @@ public readonly ref struct Context(
...
         public HashSet<ISelectionNode> Visited { get; } = new(SyntaxComparer.BySyntax);
 
+        public OrderedDictionary<string, List<FieldNode>> Fields { get; } = new(StringComparer.Ordinal);
+
         public FragmentDefinitionNode GetFragmentDefinition(string name)
             => fragments[name];
 
+        public void AddField(FieldNode field)
+        {
+            if (!Fields.TryGetValue(field.Name.Value, out var fields))
```
Comment (tobias-tengler) [inline src/HotChocolate/Fusion-vnext/src/Fusion.Execution/Planning/InlineFragmentOperationRewriter.cs:296]: Should be alias or name.

File: src/HotChocolate/Fusion-vnext/test/Fusion.Execution.Tests/Planning/InlineFragmentOperationRewriterTests.cs (hunk 1 above)
Comment (tobias-tengler) [reply src/HotChocolate/Fusion-vnext/test/Fusion.Execution.Tests/Planning/InlineFragmentOperationRewriterTests.cs:259]: Awesome, thank you! I haven't had the time to work on it myself yet. I've started on tests though, so once this is in I can take over and work on the edge cases / tests.

File: src/HotChocolate/Fusion-vnext/src/Fusion.Execution/Planning/InlineFragmentOperationRewriter.cs (hunk 2 above)
Comment (michaelstaib) [reply src/HotChocolate/Fusion-vnext/src/Fusion.Execution/Planning/InlineFragmentOperationRewriter.cs:296]: you are right
